### PR TITLE
3476

### PR DESCRIPTION
**Pull request by finkiel**
Create new report that is being sent once a week

### CODE REVIEW

File: workers/domains_without_automation/worker.go [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+	// Modify Dates
+	if nestedData, ok := data["data"].(map[string]interface{}); ok {
+		filters := nestedData["date"].(map[string]interface{})
+		if dates, ok := filters["range"].([]interface{}); ok {
+			filters["range"] = append(dates, sevenDaysAgo, today)
+		}
+	}
+	// Modify filters
+	if nestedData, ok := data["data"].(map[string]interface{}); ok {
+		filters := nestedData["filters"].(map[string]interface{})
```
Comment (denis-s-oms) [inline workers/domains_without_automation/worker.go:160]: Maybe you should cast it using also `ok` to remove chance of any panic here

File: workers/domains_without_automation/worker.go [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+			domainsForManager = append(domainsForManager, result)
+			domainsPerAccountManager[result.AccountManager] = append(domainsPerAccountManager[result.AccountManager], result)
+		}
+	}
+
+	return domainsPerAccountManager, domainsForManager
+}
+
+func createJsonForBody(domains []string) ([]byte, error) {
+	jsonFile, err := os.Open(jsonUrl)
```
Comment (denis-s-oms) [inline workers/domains_without_automation/worker.go:131]: Path for file is hardcoded. In that way you need to place json file on server by same path. Maybe make it more configurable?

File: workers/domains_without_automation/worker.go [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+const managerEmail = "Maayan Bar"
+
+type Worker struct {
+	DatabaseEnv string `json:"dbenv"`
+	Cron        string `json:"cron"`
+	httpClient  httpclient.Doer
+	skipInitRun bool
+}
+
+type TransactionIds struct {
```
Comment (denis-s-oms) [inline workers/domains_without_automation/worker.go:42]: I didn't find any usage of this struct.

File: workers/domains_without_automation/worker.go [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+	}
+
+	requestJson, err := createJsonForBody(domains)
+	if err != nil {
+		return err
+	}
+
+	results, err := fetchPubImps(token, requestJson)
+	if err != nil || len(results.Data.Result) == 0 {
+		return fmt.Errorf("error while fetching pub imps: %v", err)
```
Comment (denis-s-oms) [inline workers/domains_without_automation/worker.go:86]: If error will be nil but length of results will be 0, error message will be have no information. Something like this - `error while fetching pub imps: <nil>`

File: workers/domains_without_automation/worker.go [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+	if err != nil {
+		return nil, eris.Wrapf(err, "error creating request for compass reporting endpoint")
+	}
+	request.Header.Set("x-access-token", token)
+	request.Header.Set(fiber.HeaderContentType, fiber.MIMEApplicationJSON)
+
+	response, err := http.DefaultClient.Do(request)
+	body, err := ioutil.ReadAll(response.Body)
+	if response.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("request to report new bidder failed with status: %d", response.StatusCode)
```
Comment (denis-s-oms) [inline workers/domains_without_automation/worker.go:184]: Different approaches to error processing: somewhere it is `fmt.Errorf` and in different places it is `eris.Wrap`. I think it's better to stick to one of them.

File: workers/domains_without_automation/worker.go [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+		log.Info().Msg("Skipping work as per the skip_init_run flag.")
+		worker.skipInitRun = false
+		return nil
+	}
+
+	log.Info().Msg("Starting domains_without_automation worker")
+
+	domains, err := fetchDomainsWithoutAutomation(ctx)
+	if err != nil {
+		return err
```
Comment (denis-s-oms) [inline workers/domains_without_automation/worker.go:71]: I think it's better to add some context to this error.

File: workers/domains_without_automation/worker.go [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+		return nil, eris.Wrapf(err, "failed to marshall response body")
+	}
+
+	defer response.Body.Close()
+	return &reportResults, nil
+}
+
+func fetchToken(client httpclient.Doer, ctx context.Context) (string, error) {
+
+	pass := viper.GetString(config.TokenApiKey)
```
[the remark this answers is not in the document]
Comment (finkiel) [reply workers/domains_without_automation/worker.go:198]: your're right - changed

File: workers/domains_without_automation/worker.go (hunk 3 above)
Comment (finkiel) [reply workers/domains_without_automation/worker.go:42]: neither did I :)

**Review by denis-s-oms — APPROVED** [review]

**Pull request merged**